xcodeproj/internal/xcodeproj_rule.bzl

```diff
@@ -442,13 +442,19 @@ actual targets: {}
 
     raw_target_merge_dests = {}
     for merge in potential_target_merges:
+        if not merge.dest:
+            continue
+        
         src_target = unprocessed_targets[merge.src.id]
         src_label = bazel_labels.normalize_label(src_target.label)
         dest_target = unprocessed_targets[merge.dest]
         dest_label = bazel_labels.normalize_label(dest_target.label)
         if src_label in unfocused_labels or dest_label in unfocused_labels:
             continue
-        raw_target_merge_dests.setdefault(merge.dest, []).append(merge.src.id)
+
+        # Only include `should_create_xcode_target` targets.
+        if merge.src.id in focused_targets:
+            raw_target_merge_dests.setdefault(merge.dest, []).append(merge.src.id)
```

Comment:
I think this change stands on its own? If so, can you open a PR for just this part?

Reply:
✅  https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/1903

Reply:
Just a reminder that #1903 was reverted, so you might have to re-apply something similar.